### PR TITLE
Added details to the current_user call in the API client and improving e2e tests

### PR DESCRIPTION
**Pull request by kiddinn**
This PR adds few things:

1. A new user object that calls the `users/me/` API resource and interprets those results.
2. More fields returned in the `users/me/` API resource
3. The client API call for `current_user` now returns the user object.
4. Improving e2e tests, adding functionality and new tests.

The E2E test improvements are:

+ Added a status check to see if the search index is ready to prevent tests to start before timelines are fully ingested
+ Adding a new search test, with more options to the search object.
+ Adding the ability to run an analyzer on an ingested timeline in e2e tests
+ Added few client tests.
+ Made few fixes here and there after some of the e2e tests revealed them.

### CODE REVIEW

**Comment by kiddinn** [comment]
moved this back to draft, since I'm adding few e2e tests and that required some additional changes to the codebase

**Comment by kiddinn** [comment]
@berggren this is ready now for a review

File: api_client/python/timesketch_api_client/client.py [hunk 1]
```diff
@@ -507,7 +507,11 @@ def list_searchindices(self):
         """
         indices = []
         response = self.fetch_resource_data('searchindices/')
-        for index_dict in response['objects'][0]:
+        objects = response.get('objects')
+        if not objects:
+            return indices
+
+        for index_dict in objects[0]:
```
Comment (berggren) [inline api_client/python/timesketch_api_client/client.py:514]:
```suggestion
        response_objects = response.get('objects')
        if not response_objects:
            return indices

        for index_dict in response_objects[0]:
```

**Review by berggren — APPROVED** [review]
Just a small nit, feel free to ignore. Otherwise LGTM

**Pull request merged**